integration_tests/steps/myco.py

```diff
@@ -41,7 +41,7 @@ def step_impl(context):
     # placing bids and offers on every market cycle.
     # Should stop if an error occurs or if the simulation has finished
     counter = 0
-    while context.matcher.errors == 0 and not context.matcher.is_finished:
+    while context.matcher.errors == 0 and not context.matcher.is_finished and counter < 150:
```

Comment:
Could you please explain what this counter is for? In order to stop the test simulations we usually use the `--duration` parameter, right?

Reply:
Indeed, however we cannot depend on it
Let's assume that the d3a raised an exception and stopped, the myco won't stop unless it receives an error (as in the integration test class) or is_finished (the d3a has to send a `finish` event), so if the d3a broke, we will end up with an infinite test time in this side